cpp/src/arrow/compute/exec/spilling_join.cc

```diff
@@ -0,0 +1,351 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/compute/exec/spilling_join.h"
+#include "arrow/util/atomic_util.h"
+#include "arrow/util/make_unique.h"
+
+namespace arrow
+{
+    namespace compute
+    {
+        void PartitionedBloomFilter::Find(
+                int64_t hardware_flags,
+                int64_t num_rows,
+                const uint64_t *hashes,
+                uint8_t *bv)
+        {
+            if(in_memory)
+                return in_memory->Find(hardware_flags, num_rows, hashes, bv);
+
+            for(int64_t i = 0; i < num_rows; i++)
+            {
+                uint64_t hash = hashes[i];
+                size_t partition = hash & (SpillingAccumulationQueue::kNumPartitions - 1);
```

Comment:
hash&(kNumPartitions-1) ->This formula is also used in SpillingAccumulationQueue. This needs to be made into a separate method with a comment saying that we REALLY need to remember to use the lowest bits of hash and not the highest and why (because hash table code uses the highest bits and correlation with partition selection would result in high number of hash collisions).

Reply:
Pulled it out into a static method of SpillingAccumulationQueue and added comment.